### PR TITLE
Studio mesh reconstruction

### PR DESCRIPTION
**Pull request by akenmorris**
This PR addresses the following issues:

#914 - Mesh and particles disappear in studio after optimization
#913 - Studio: automatic glyph size 
#911 - Studio: Allow choice of surface reconstruction method based on data
#902 - Studio: optimize tool should load images in the worker thread
#822 - Studio should auto-run the surface reconstruction like View2 does
#870 - Studio: Drag and Drop image volumes and meshes to import 
#844 - Mesh reconstruction via template transform to studio for mesh support 
#917 - Studio: Particle update interval needs to be tuned based on refresh duration
#918 - Studio: Display filenames for old XML inputs

studio_analyze_surface_reconstruction

### CODE REVIEW

**Comment by medakk** [comment]
This is great stuff!! :D 


I noticed this weird behaviour: after optimizing, if I go to analyze, it seems to be using mesh-warping based reconstruction but the radio button says its using legacy:
image

**Comment by akenmorris** [comment]
@medakk , yes indeed.  Will push fixes shortly.

File: Libs/Mesh/Mesh.h [hunk 1]
```diff
@@ -14,8 +15,10 @@ class Mesh
   using MeshType = vtkSmartPointer<vtkPolyData>;
 
   Mesh(const std::string& pathname) : mesh(read(pathname)) {}
-  Mesh(vtkSmartPointer<vtkPolyData>&& rhs) : mesh(std::move(rhs)) {}
-
+  // Mesh(vtkSmartPointer<vtkPolyData>&& rhs) : mesh(std::move(rhs)) {}
```
Comment (cchriste) [inline Libs/Mesh/Mesh.h:18]: Please see my comments about this in the mesh warp pull request.

**Review by cchriste — CHANGES_REQUESTED** [review]
Looks mostly okay. Just the pieces from the mesh warp pull request (modifications to Mesh constructors and assignment operators, etc., just see the comments over there). Otherwise, great.

I suggest merging it with the new alpha branch, thus enabling testing the consolidated mesh work from three branches before springing it on an unsuspecting master branch.

**Pull request closed without merging**